QtAcademyProject.h

```diff
@@ -27,20 +36,46 @@ class QtAcademyProject : public QObject
   Q_OBJECT
 
   Q_PROPERTY(Esri::ArcGISRuntime::MapQuickView* mapView READ mapView WRITE setMapView NOTIFY mapViewChanged)
+  Q_PROPERTY(int downloadProgress READ downloadProgress NOTIFY downloadProgressChanged)
+  Q_PROPERTY(bool isTracking READ isTracking WRITE setIsTracking NOTIFY isTrackingChanged)
 
 public:
   explicit QtAcademyProject(QObject* parent = nullptr);
   ~QtAcademyProject() override;
 
+  Q_INVOKABLE void createOfflineAreaFromExtent();
+  Q_INVOKABLE void toggleOffline(bool offline);
+
 signals:
   void mapViewChanged();
+  void downloadProgressChanged();
+  void isTrackingChanged();
 
 private:
   Esri::ArcGISRuntime::MapQuickView* mapView() const;
   void setMapView(Esri::ArcGISRuntime::MapQuickView* mapView);
+  void loadOfflineBasemaps();
+  void exportVectorTiles(Esri::ArcGISRuntime::ArcGISVectorTiledLayer* vectorTileLayer);
+  int downloadProgress();
+  bool isTracking() const;
+  void setIsTracking(bool value);
 
   Esri::ArcGISRuntime::Map* m_map = nullptr;
   Esri::ArcGISRuntime::MapQuickView* m_mapView = nullptr;
+
+  Esri::ArcGISRuntime::OfflineMapTask* m_offlineMapTask = nullptr;
+  Esri::ArcGISRuntime::ExportVectorTilesTask* m_exportVectorTilesTask = nullptr;
+  Esri::ArcGISRuntime::Basemap* m_basemap = nullptr;
+
+  Esri::ArcGISRuntime::Graphic* m_offlineMapExtentGraphic = nullptr;
+  Esri::ArcGISRuntime::ExportVectorTilesJob* m_exportVectorTilesJob = nullptr;
+  Esri::ArcGISRuntime::VectorTileCache* m_vectorTileCache = nullptr;
+  Esri::ArcGISRuntime::ItemResourceCache* m_itemResourceCache = nullptr;
+  Esri::ArcGISRuntime::ArcGISVectorTiledLayer* m_offlineLayer = nullptr;
+  Esri::ArcGISRuntime::PolylineBuilder* m_lineBuilder = nullptr;
+  int m_downloadProgress = 0;
+  bool m_isTracking = false;
+  std::unique_ptr<QObject> m_tempObject;
```

Comment:
```suggestion
  std::unique_ptr<QObject> m_tempParent;
```
Similar changes also needed in the .cpp file. IMO "parent" conveys its purpose better than "object".